### PR TITLE
tmpdir refactor

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Changed:
- `tmpdir` for all variant calling rules

Removed:
- Removed unused rules and cleaned up commented out lines.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by hassanfa** [comment]
Ashwini, I started to work on splitting VCF and noticed it'd be easier to work on a cleaned up and refactored rule first.

File: BALSAMIC/snakemake_rules/variant_calling/germline.rule [hunk 1]
```diff
@@ -36,52 +34,19 @@ rule haplotypecaller_merge:
...
     benchmark:
         benchmark_dir + 'haplotypecaller_merge_' + "SNV.germline.{sample}.haplotypecaller.vcf.gz" 
     shell:
         "source activate {params.conda}; "
-        "rand_str=$(openssl rand -hex 5); "
-        "tmpdir={params.tmpdir}/${{rand_str}}; "
-        "mkdir -p ${{tmpdir}}; "
-        "export TMPDIR=${{tmpdir}}; "
+        "mkdir -p {params.tmpdir}; "
+        "export TMPDIR={parmams.tmpdir}; "
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/germline.rule:45]:
```suggestion
        "export TMPDIR={params.tmpdir}; "
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_t_varcall.rule [hunk 2]
```diff
@@ -33,11 +33,9 @@ rule sentieon_TNsnv_tumor_only:
         benchmark_dir + config["analysis"]["case_id"] + ".tnsnv.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_t_varcall.rule:37]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_t_varcall.rule [hunk 3]
```diff
@@ -103,39 +99,12 @@ rule sentieon_TNscope_tumor_only:
         benchmark_dir + config["analysis"]["case_id"] + ".tnscope.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_t_varcall.rule:103]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 4]
```diff
@@ -13,7 +13,7 @@ rule sentieon_TN_corealign:
     output:
         bam = bam_dir + config["analysis"]["case_id"] + ".corealign.bam"
     params:
-        tmpdir = tmp_dir,
+        tmpdir = tempfile.mkstemp(prefix=tmp_dir, suffix="sentieon_TN_corealign"+config["analysis"]["case_id"]),
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:16]:
```suggestion
        tmpdir = tempfile.mkstemp(prefix=tmp_dir, suffix="sentieon_TN_corealign" + config["analysis"]["case_id"]),
```

File: BALSAMIC/snakemake_rules/variant_calling/cnvkit_paired.rule [hunk 5]
```diff
@@ -34,7 +34,7 @@ rule cnvkit_paired:
         gene_metrics = cnv_dir + config["analysis"]["case_id"] + ".gene_metrics",
     params:
         housekeeper_id = {"id": config["analysis"]["case_id"], "tags": "cnv"},
-        tmpdir = tmp_dir,
+        tmpdir = tempfile.mkstemp(prefix=tmp_dir, suffix="cnvkit_paired_"+config["analysis"]["case_id"]),
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/cnvkit_paired.rule:37]:
```suggestion
        tmpdir = tempfile.mkstemp(prefix=tmp_dir, suffix="cnvkit_paired_" + config["analysis"]["case_id"]),
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 6]
```diff
@@ -23,11 +23,9 @@ rule sentieon_TN_corealign:
         benchmark_dir + 'sentieon_TN_corealign_' + config["analysis"]["case_id"] + ".corealign.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:27]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 7]
```diff
@@ -91,11 +87,9 @@ rule sentieon_TNhaplotyper:
         benchmark_dir + 'sentieon_TNhaplotyper_' + config["analysis"]["case_id"] + ".tnhaplotyper.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:91]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 8]
```diff
@@ -160,11 +152,9 @@ rule sentioen_filter_TNscope:
         benchmark_dir + 'sentioen_filter_TNscope_' + config["analysis"]["case_id"] + ".tnscope_filter.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:156]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 9]
```diff
@@ -129,11 +123,9 @@ rule sentieon_TNscope:
         benchmark_dir + 'sentieon_TNscope_' + config["analysis"]["case_id"] + ".tnscope.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:127]:
```suggestion
export TMPDIR={params.tmpdir};
```

File: BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule [hunk 10]
```diff
@@ -57,11 +55,9 @@ rule sentieon_TNsnv:
         benchmark_dir + 'sentieon_TNsnv_' + config["analysis"]["case_id"] + ".tnsnv.tsv"
     shell:
         """
-rand_str=$(openssl rand -hex 5);
-tmpdir={params.tmpdir}/${{rand_str}};
-mkdir -p ${{tmpdir}};
-export TMPDIR=${{tmpdir}};
-export SENTIEON_TMPDIR=${{tmpdir}};
+mkdir -p {params.tmpdir};
+export TMPDIR={parms.tmpdir};
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/sentieon_tn_varcall.rule:59]:
```suggestion
export TMPDIR={params.tmpdir};
```

**Review by ashwini06 — COMMENTED** [review]
Just few comments on spell checks (`parms`->`params` ). 

How these `temp_dirs` are handled in the end? is it automatically deleted at the end of the workflow? 

As you are going through these rules, do you also wanna restructure the snakemake rule structure as discussed here #406

**Comment by hassanfa** [comment]
`shutils.rmtree(tmpdir) in `rule all:`  to delete all the temporary files.

I'm trying to adhere to that definition and fix as many as possible.

**Comment by hassanfa** [comment]
Testing run for refactored rules. I'll update results as soon as it is finished.

**Comment by hassanfa** [comment]
Analysis is finished and ready to merge!
image

**Pull request merged**